sdk/identity/azure-identity/azure/identity/_credentials/browser.py

```diff
@@ -56,78 +53,30 @@ def __init__(self, **kwargs) -> None:
 
         self._login_hint = kwargs.pop("login_hint", None)
         self._timeout = kwargs.pop("timeout", 300)
-        self._server_class = kwargs.pop("_server_class", AuthCodeRedirectServer)
         client_id = kwargs.pop("client_id", DEVELOPER_SIGN_ON_CLIENT_ID)
         super(InteractiveBrowserCredential, self).__init__(client_id=client_id, **kwargs)
 
     @wrap_exceptions
     def _request_token(self, *scopes: str, **kwargs) -> Dict:
-
-        # start an HTTP server to receive the redirect
-        server = None
-        if self._parsed_url:
-            try:
-                redirect_uri = "http://{}:{}".format(self._parsed_url.hostname, self._parsed_url.port)
-                server = self._server_class(self._parsed_url.hostname, self._parsed_url.port, timeout=self._timeout)
-            except socket.error:
-                raise CredentialUnavailableError(message="Couldn't start an HTTP server on " + redirect_uri)
-        else:
-            for port in range(8400, 9000):
-                try:
-                    server = self._server_class("localhost", port, timeout=self._timeout)
-                    redirect_uri = "http://localhost:{}".format(port)
-                    break
-                except socket.error:
-                    continue  # keep looking for an open port
-
-        if not server:
-            raise CredentialUnavailableError(message="Couldn't start an HTTP server on localhost")
-
-        # get the url the user must visit to authenticate
         scopes = list(scopes)  # type: ignore
         claims = kwargs.get("claims")
         app = self._get_app(**kwargs)
-        flow = app.initiate_auth_code_flow(
-            scopes,
-            redirect_uri=redirect_uri,
-            prompt="select_account",
-            claims_challenge=claims,
-            login_hint=self._login_hint,
-        )
-        if "auth_uri" not in flow:
-            raise CredentialUnavailableError("Failed to begin authentication flow")
-
-        if not _open_browser(flow["auth_uri"]):
-            raise CredentialUnavailableError(message="Failed to open a browser")
-
-        # block until the server times out or receives the post-authentication redirect
-        response = server.wait_for_redirect()
-        if not response:
-            raise ClientAuthenticationError(
-                message="Timed out after waiting {} seconds for the user to authenticate".format(self._timeout)
+        port = self._parsed_url.port if self._parsed_url else None
+
+        try:
+            result = app.acquire_token_interactive(
+                scopes=scopes,
+                login_hint=self._login_hint,
+                claims_challenge=claims,
```

Comment:
Should we also propagate timeout?
```suggestion
                claims_challenge=claims,
                timeout=self._timeout,
```

Reply:
Though after testing the timeout, the timeout works, but msal doesn't seem to give a proper exception/error message that indicates a timeout...
```
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/application.py", line 1828, in acquire_token_interactive
    response = _clean_up(self.client.obtain_token_by_browser(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/oauth2cli/oidc.py", line 281, in obtain_token_by_browser
    return super(Client, self).obtain_token_by_browser(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/oauth2cli/oauth2.py", line 639, in obtain_token_by_browser
    return self._obtain_token_by_browser(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/oauth2cli/oauth2.py", line 677, in _obtain_token_by_browser
    return self.obtain_token_by_auth_code_flow(
  File "/home/pvaneck/dev/pvaneck/azure-sdk-for-python/sdk/identity/azure-identity/.venv/lib/python3.10/site-packages/msal/application.py", line 134, in obtain_token_by_auth_code_flow
    assert isinstance(auth_code_flow, dict) and isinstance(auth_response, dict)
AssertionError
```

Reply:
@pvaneck Thanks for your finding. This is a bug in MSAL. Interesting that nobody reported it before, so I guess nobody really uses the "timeout" behavior.

I'll fix it soon, probably by throwing something like [TimeoutError](https://docs.python.org/3/library/exceptions.html#TimeoutError).

CC: @xiangyan99

Reply:
@rayluo what is current error?

You may need to evaluate if it is a breaking change.

Reply:
Thanks @rayluo. I opened an issue just recently to track it: https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/516